src/sinks/aws_cloudwatch_metrics.rs

```diff
@@ -117,11 +117,10 @@ impl CloudWatchMetricsSvc {
         client: CloudWatchClient,
         cx: SinkContext,
     ) -> crate::Result<super::RouterSink> {
-        let batch = config
-            .batch
-            .disallow_max_bytes()?
-            .use_size_as_events()?
-            .get_settings_or_default(BatchSettings::default().events(20).timeout(1));
+        let batch = BatchSettings::default()
+            .events(20)
```

Comment:
Do you think it would be beneficial to conventionalize having these defaults as some `const` values inside sinks that use them?

Reply:
Oh yeah, since all the methods are `const`, these could totally get moved out of the function.

Reply:
Um, actually, no they can't, or at least not without repeating the buffer type in the const declaration. The actual type of these settings are `BatchSettings<B> where B: Batch`, and the `B` is inferred by passing the settings to the buffer creation function. This allows the batch parsing to be controlled by the buffer type, which was the whole point of this PR.

Reply:
Totally makes sense!